tests/lava/lib/dl/slayer/block/test_cuba.py

```diff
@@ -25,27 +26,42 @@
 torch.manual_seed(seed)
 torch.cuda.manual_seed(seed)
 if verbose:
-    print(f'{seed=}')
+    print(f"{seed=}")
 
 if torch.cuda.is_available():
-    device = torch.device('cuda')
+    device = torch.device("cuda")
 else:
     if verbose:
-        print('CUDA is not available in the system. '
-              'Testing for CPU version only.')
-    device = torch.device('cpu')
+        print(
+            "CUDA is not available in the system. "
+            "Testing for CPU version only."
+        )
+    device = torch.device("cpu")
 
-tempdir = os.path.dirname(__file__) + '/temp'
+tempdir = os.path.dirname(__file__) + "/temp"
 os.makedirs(tempdir, exist_ok=True)
 
-neuron_param = {'threshold': 0.5,
-                'current_decay': 0.5,
-                'voltage_decay': 0.5}
+neuron_param = {"threshold": 0.5, "current_decay": 0.5, "voltage_decay": 0.5}
 
 
 class TestCUBA(unittest.TestCase):
     """Test CUBA blocks"""
 
+    def test_affine_block_hdf5_export(self):
+        """Test affine block hdf5 export."""
+        in_features = 10
+        out_features = 5
+
+        # create slayer network and evaluate output
+        net = slayer.block.cuba.Affine(neuron_param, in_features, out_features)
+
+        # export slayer network
+        net.export_hdf5(
+            h5py.File(tempdir + "/cuba_affine.net", "w").create_group(
+                "layer/0"
+            )
+        )
```

Comment:
This is not a proper unittest. Can you make it more elaborate?

Reply:
The test fails if the network consisting of  an Affine layer is not exportable. Should I catch a specific error? I could construct an "assert" statement somehow, but I don't think this will improve the test. But maybe I am missing something here, could you point me to it?

Reply:
@ahenkes1 I get that, can you also reload the hdf5 file and assert that the layer neuron type is Affine or something like that.

Reply:
the `net_str` of the layer after importing using `netx.hdf5.Network()` says the layer is of type `dense`. This is consistent with the supported layer types reported in the [netx documentation](https://lava-nc.org/lava-lib-dl/netx/netx.html).

Does that mean, the import is working? In that case, I would have to check, if the output of the original and the netx model are the same.

On the other hand, if it means that the import is not working, the Affine layer has to be included into the supported layers with all its consequences.

@bamsumit , could you clarify on that?

Reply:
@ahenkes1 affine layer has two modes.
1. By default, the `dynamics=True` which means that the neuron dynamics is applied, but the neuron does not spike. In this case, you should export the neuron parameters in the hdf5 config like below in `src/lava/lib/dl/slayer/block/base.py: line394`
    ```python
        if self.dynamics is True:
            for key, value in self.neuron.device_params.items():
                if key == 'vThMant':
                    value = (1 << 17) - 1  # set the maximum possible threshold
                handle.create_dataset(f'neuron/{key}', data=value)
    ```
    __In this case, you can test `layer['neuron']['vThMant'] == (1 << 17) - 1` holds or not__
2. If `dynamics=False`, the output is the output of the synaptic operation. The above piece of code will not export the Neuron parameters. In that case, it will be automatically be assigned a generic CUBA neuron which just sets the synapse output in its `current` state (`src/lava/lib/dl/netx/hdf5.py: line 330`).
__In this case, you can test `'neuron' not in layer.keys()` holds or not__

Reply:
@bamsumit thanks for the answer and imsifhts, I will adjust the test accordingly!

Reply:
@bamsumit I constructed two unit tests which both currently fail. 

As far as I understand, for scenario `1. dynamic=True` the parameter export has to be manually implemented for all parameters. If that is the case, I can do that. 

For the second scenario, `dynamic=False`, everything should work out of the box, which is unfortunately not the case. Taking a look at `src/lava/lib/dl/netx/hdf5.py: line 326-338`

```
        if 'neuron' in layer_config.keys():
            neuron_config = layer_config['neuron']
        else:
            # Non-leaky integrator by default
            neuron_config = {'type': 'CUBA',
                             'iDecay': 0,
                             'vDecay': 4096,
                             'vThMant': 1 << 18 - 1,
                             'refDelay': 1,
                             'gradedSpike': False}
        neuron_params = Network.get_neuron_params(neuron_config,
                                                  reset_interval=reset_interval,
                                                  reset_offset=reset_offset)
```
the `else` branch should be followed, as you pointed out. Indeed, this branch is entered while exporting the Affine layer. But this will assign a 'neuron' to the layer and your suggested test fails.  Honestly, I am a little bit stuck on this one. Maybe you can take a look at the test and provide a possible direction.

Reply:
@ahenkes1 
1. In scenario 1, you can add the following line in `src/lava/lib/dl/slayer/block/base.py: line394`
```python
    if self.dynamics is True:
        for key, value in self.neuron.device_params.items():
            if key == 'vThMant':
                value = (1 << 17) - 1  # set the maximum possible threshold
            handle.create_dataset(f'neuron/{key}', data=value)
```
2. In scenario 2, you test `'neuron' in lava_net.net_config['layer'][0].keys()`

BTW, please refrain from ussing `assert` like `assert 'neuron' not in layer.__dict__.keys()`. Instead use `self.assertTrue(...)`